packages/ncids-js/src/components/nci-autocomplete/nci-autocomplete.component.ts

```diff
@@ -0,0 +1,523 @@
+	 * Instantiates this component of the given element.
+	 *
+	 * @param {HTMLElement} element Component being created.
+	 * @param {NCIExtendedHeaderWithMegaMenuOptions} options Optional settings for component generation.
+	 * @public
+	 */
+	public static create(
+		element: HTMLInputElement,
+		options: {
+			autocompleteSource: AutocompleteAdaptor;
+		} & Partial<AutocompleteOptions>
+	): NCIAutocomplete {
+		if (!(element instanceof HTMLInputElement)) {
+			throw 'Must be an input element to be an autocomplete';
+		}
+		return new this(element, options);
+	}
+
+	/**
+	 * Finds collapsible sections, the buttons that triggers them, and adds event
+	 * listeners and custom events.
+	 * @private
+	 */
+	private initialize(): void {
+		this.updateDom();
+	}
+
+	/**
+	 * Initial setup of listbox, announcer, and other supporting elements and attributes
+	 * @private
+	 */
+	private updateDom(): void {
+		//set up listbox
+		this.listboxWrapper.setAttribute(
+			'id',
+			this.autocompleteInputId + '-termswrapper'
+		);
+		this.listboxWrapper.classList.add('nci-autocomplete__listbox');
+		if (this.options.listboxClasses !== '') {
+			this.listboxWrapper.classList.add(this.options.listboxClasses);
+		}
+		this.listbox.setAttribute('id', this.autocompleteInputId + '-terms');
+		this.listbox.setAttribute('tabindex', '-1');
+		this.listbox.setAttribute('role', 'listbox');
+		this.listboxWrapper.append(this.listbox);
+		this.autocompleteContainer.append(this.listboxWrapper);
+
+		//set up announcer
+		this.announcer.classList.add('nci-autocomplete__status');
+		this.announcer.setAttribute('aria-live', 'assertive');
+		this.autocompleteContainer.prepend(this.announcer);
+
+		// a11y-ify input
+		this.newACInput.setAttribute('role', 'combobox');
+		this.newACInput.setAttribute('aria-autocomplete', 'list');
+		this.newACInput.setAttribute(
+			'aria-owns',
+			this.autocompleteInputId + '-terms'
+		);
+		this.newACInput.setAttribute('aria-activedescendant', '');
+
+		this.addEventListeners();
+	}
+
+	/**
+	 * Sets up form and input event listeners
+	 * @private
+	 */
+	private addEventListeners(): void {
+		this.newACInput.addEventListener('input', this.inputEventListener, true);
+		this.newACInput.addEventListener(
+			'keydown',
+			this.keyPressEventListener,
+			true
+		);
+		this.acForm.addEventListener('submit', this.submissionEventListener, true);
+		// if not clicking on active listbox, close it
+		window.addEventListener('click', this.outsideClickListener, true);
+	}
+
+	/**
+	 * Utility to hide listbox
+	 * @private
+	 */
+	private closeListbox(): void {
+		this.newACInput.removeAttribute('aria-activedescendant');
+		this.announcer.innerHTML = '';
+		this.listbox.innerHTML = '';
+		this.listboxWrapper.classList.remove('active');
+	}
+
+	private handleFormSubmission(): void {
+		const formDetail = {
+			searchText: this.newACInput.value,
+			optionsPresented: this.optionsListDisplayed,
+			...this.selectedOptionInfo,
+		};
+		window.dispatchEvent(
```

Comment:
Can you please change this to dispatch on the input element. This way we can handle multiple autocompletes in a form. 

Also nci-autocomplete is mispelled.